src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -109,13 +105,15 @@ internal abstract partial class AbstractAddImportFeatureService<TSimpleNameSynta
                                 document, semanticModel, diagnosticId, node, maxResults, symbolSearchService,
                                 options, packageSources, cancellationToken).ConfigureAwait(false);
 
-                            // Nothing found at all. No need to proceed.
                             foreach (var reference in allSymbolReferences)
                             {
                                 cancellationToken.ThrowIfCancellationRequested();
 
                                 var fixData = await reference.TryGetFixDataAsync(document, node, options.CleanupOptions, cancellationToken).ConfigureAwait(false);
                                 result.AddIfNotNull(fixData);
+
+                                if (result.Count > maxResults)
+                                    break;
```

Comment:
Moved this logic here as a much simpler place to check, versus in the concurrent searching we're doing.